src/render/draw_raster-data-driven.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+
+import { clamp } from '../util/util';
+
+import ImageSource from '../source/image_source';
+import browser from '../util/browser';
+import StencilMode from '../gl/stencil_mode';
+import DepthMode from '../gl/depth_mode';
+import CullFaceMode from '../gl/cull_face_mode';
+import { rasterDataDrivenUniformValues } from './program/raster-data-driven_program';
+import Texture from '../render/texture';
+
+
+import type Painter from './painter';
+import type SourceCache from '../source/source_cache';
+import type RasterDataDrivenStyleLayer from '../style/style_layer/raster-data-driven_style_layer';
+import type {OverscaledTileID} from '../source/tile_id';
+
+export default drawRaster;
+
+function drawRaster(painter: Painter, sourceCache: SourceCache, layer: RasterDataDrivenStyleLayer, coords: Array<OverscaledTileID>) {
+    if (painter.renderPass !== 'translucent') return;
+    if (layer.paint.get('raster-opacity') === 0) return;
+
+    const context = painter.context;
+    const gl = context.gl;
+    const source = sourceCache.getSource();
+    const program = painter.useProgram('rasterDataDriven');
+
+    const stencilMode = StencilMode.disabled;
+    const colorMode = painter.colorModeForRenderPass();
+    const minTileZ = coords.length && coords[0].overscaledZ;
+
+    // ubilabs start
+    const imageData = layer._lookupTexture;
+    const lookupTexture = new Texture(context, imageData, gl.RGBA);
+
+    context.activeTexture.set(gl.TEXTURE2);
+    lookupTexture.bind(gl.NEAREST, gl.CLAMP_TO_EDGE);
+    // ubilabs end
+
+    for (const coord of coords) {
+        // Set the lower zoom level to sublayer 0, and higher zoom levels to higher sublayers
+        // Use gl.LESS to prevent double drawing in areas where tiles overlap.
+        const depthMode = painter.depthModeForSublayer(coord.overscaledZ - minTileZ,
+            layer.paint.get('raster-opacity') === 1 ? DepthMode.ReadWrite : DepthMode.ReadOnly, gl.LESS);
+
+        const tile = sourceCache.getTile(coord);
+        const posMatrix = painter.transform.calculatePosMatrix(coord.toUnwrapped(), true);
+
+        tile.registerFadeDuration(layer.paint.get('raster-fade-duration'));
+
+        const parentTile = sourceCache.findLoadedParent(coord, 0),
+            fade = getFadeValues(tile, parentTile, sourceCache, layer, painter.transform);
+
+        let parentScaleBy, parentTL;
+
+        const textureFilter = layer.paint.get('raster-resampling') === 'nearest' ?  gl.NEAREST : gl.LINEAR;
```

Comment:
The `data-driven-raster` should turn this around: default to `NEAREST`.

Reply:
You mean like this: `layer.paint.get('raster-resampling') === 'linear' ? gl.LINEAR : gl.NEAREST`

Reply:
yep!